archook/archook.py

```diff
@@ -7,56 +7,75 @@
+
   HLKM/SOFTWARE/ESRI/ArcGIS 'RealVersion' - will give the version, then we can use
   that to go to
   HKLM/SOFTWARE/ESRI/DesktopXX.X 'InstallDir'. Where XX.X is the version
 
   We may need to check HKLM/SOFTWARE/Wow6432Node/ESRI instead
   '''
   try:
-    key = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE,
-                          'SOFTWARE\\Wow6432Node\\ESRI\\ArcGIS', 0)
+    if pro:
+      pro_key = _winreg.OpenKey(
+        _winreg.HKEY_LOCAL_MACHINE,
+        'SOFTWARE\\ESRI\\ArcGISPro'
+      )
+      install_dir = _winreg.QueryValueEx(pro_key, "InstallDir")[0]
+    else:
+      key = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE,
+                            'SOFTWARE\\Wow6432Node\\ESRI\\ArcGIS', 0)
 
-    version = _winreg.QueryValueEx(key, "RealVersion")[0][:4]
+      version = _winreg.QueryValueEx(key, "RealVersion")[0][:4]
 
-    key_string = "SOFTWARE\\Wow6432Node\\ESRI\\Desktop{0}".format(version)
-    desktop_key = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE,
-                                  key_string, 0)
+      key_string = "SOFTWARE\\Wow6432Node\\ESRI\\Desktop{0}".format(version)
+      desktop_key = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE,
+                                    key_string, 0)
 
-    install_dir = _winreg.QueryValueEx(desktop_key, "InstallDir")[0]
+      install_dir = _winreg.QueryValueEx(desktop_key, "InstallDir")[0]
     return install_dir
   except WindowsError:
     raise ImportError("Could not locate the ArcGIS directory on this machine")
 
-def get_arcpy():  
+def get_arcpy(pro=False):  
   '''
   Allows arcpy to imported on 'unmanaged' python installations (i.e. python installations
   arcgis is not aware of).
   Gets the location of arcpy and related libs and adds it to sys.path
+  Looks for ArcGIS Pro if `pro` argument is True.
   '''
-  install_dir = locate_arcgis()  
-  arcpy = path.join(install_dir, "arcpy")
-  # Check we have the arcpy directory.
-  if not path.exists(arcpy):
-    raise ImportError("Could not find arcpy directory in {0}".format(install_dir))
-
-  # First check if we have a bin64 directory - this exists when arcgis is 64bit
-  bin_dir = path.join(install_dir, "bin64")
-  
-  # check if we are using a 64-bit version of Python
-  is_64bits = sys.maxsize > 2**32
-  
-  if not path.exists(bin_dir) or is_64bits == False:
-    # Fall back to regular 'bin' dir otherwise.
-    bin_dir = path.join(install_dir, "bin")
-
-  scripts = path.join(install_dir, "ArcToolbox", "Scripts")  
-  sys.path.extend([arcpy, bin_dir, scripts])
+  install_dir = locate_arcgis(pro)
+  if pro:
+    os.environ['PATH'] = ';'.join((os.path.join(install_dir, 'bin'), os.environ['PATH']))
+    sys.path.append(os.path.join(install_dir, 'bin'))
+    sys.path.append(os.path.join(install_dir, r'Resources\ArcPy'))
+    sys.path.append(os.path.join(install_dir, r'Resources\ArcToolbox\Scripts'))
+    sys.path.append(os.path.join(install_dir, r'bin\Python\envs\arcgispro-py3\Lib\site-packages'))
```

Comment:
Conda managed env may not be at this location reliably. (I would still accept the PR, this is more a note for later.)

Reply:
@mhw-at-yg thanks for taking a look at this!

Would the location specified by the `PythonCondaRoot` and `PythonCondaEnv` keys under `HKEY_LOCAL_MACHINE\SOFTWARE\ESRI\ArcGISPro` be reliable?

e.g. `%PythonCondaRoot%\envs\%PythonCondaEnv%\Lib\site-packages` would expand to `C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\Lib\site-packages` on my PC.

Reply:
I believe *Root and *Env should be reliable, and agree getting into user-specific envs is probably not warranted until someone comes along needing it (preferably with code ;-)

Reply:
@mhw-at-yg added the registry lookup for the ArcGIS Pro managed conda environment.

Reply:
As far as I can see, now it will *only* add site-packages if they are in a conda env and if conda is not being used, an ``ImportError`` will be thrown - is this correct?
I don't know but I would assume conda is 'opt in' and not necessarily required by arcgis pro?

Reply:
@JamesRamm conda is installed and required by ArcGIS Pro, that is, unless you are using `archook`, then you can use a standalone Python 3.6 interpreter.